packages/http/src/mocker/callback/scheduler.ts

```diff
@@ -0,0 +1,45 @@
+import { IHttpCallbackOperation, IHttpOperationRequest } from '@stoplight/types';
+import { IHttpOperationConfig, IHttpRequest, IHttpResponse } from '../../types';
+import withLogger from '../../withLogger';
+import { runCallback } from './runner';
+import { RangeOrNumber } from '@stoplight/prism-core';
+import { pipe } from 'fp-ts/lib/pipeable';
+import * as Option from 'fp-ts/lib/Option';
+
+export function scheduleCallback({
+                                   callback,
+                                   request,
+                                   response,
+                                   config,
+                                 }: {
+  callback: IHttpCallbackOperation;
+  request: IHttpRequest;
+  response: IHttpResponse;
+  config: Pick<IHttpOperationConfig, 'callbackDelay' | 'callbackCount'>
+}) {
+  return withLogger(logger => {
+    let executionsLeft = reduceRange(config.callbackCount || 1);
+
+    const execute = () => {
+      if (executionsLeft > 0) {
+        setTimeout(() => {
+          executionsLeft--;
+
+          runCallback({ callback, request, response })(logger)()
+            .finally(execute);
+        }, reduceRange(config.callbackDelay || 0));
+      }
+    };
+
+    execute();
+  });
+}
```

Comment:
I need to think a little bit more about this — the reader should be propagated outside and not resolved here.